nexus/internal/filetransfer/file_transfer_manager.go

```diff
@@ -202,13 +204,25 @@ func (fm *FileTransferManager) transfer(task *Task) error {
 	case UploadTask:
 		err = fm.fileTransfer.Upload(task)
 	case DownloadTask:
+		if err := fm.ensureDownloadRootDir(task.Path); err != nil {
+			return err
+		}
 		err = fm.fileTransfer.Download(task)
 	default:
 		fm.logger.CaptureFatalAndPanic("sender: sendRecord: nil RecordType", err)
 	}
 	return err
 }
 
+func (fm *FileTransferManager) ensureDownloadRootDir(filePath string) error {
+	baseDir := path.Dir(filePath)
+	info, err := os.Stat(baseDir)
+	if err == nil && info.IsDir() {
+		return nil
+	}
+	return os.MkdirAll(baseDir, 0777)
```

Comment:
Hmm shouldn't we respect the umask here as well? Btw, I'm surprised Go asks for the permissions instead of using the default.

Reply:
https://stackoverflow.com/questions/14249467/os-mkdir-and-os-mkdirall-permissions

The final set of permissions that the directory receives is after the umask permissions are removed.